pylint/lint/run.py

```diff
@@ -175,9 +175,11 @@ def __init__(
                 sys.exit(code)
             return
 
-        # Display help if there are no files to lint or no checks enabled
-        if not args or len(linter.config.disable) == len(
-            linter.msgs_store._messages_definitions
+        # Display help if there are no files to lint or only internal checks enabled (`--disable=all`)
+        if not args or (
+            len(linter.config.enable) == 0
+            and len(linter.config.disable)
+            == len(linter.msgs_store.messages) - len(linter.default_enabled_messages)
```

Comment:
I think we should do a more exact comparison. This will accidentally catch the case where someone manually disables exactly 401 messages, including some of the default enabled ones.

Maybe it's enough to also test that there is not an intersection between the disable and the default_enabled_messages.